### PR TITLE
DW-3019 update hdi logging calls - intermediate updates to keep the PR sizes down

### PR DESCRIPTION
**Pull request by mark-r-m**
DW-3019 update hdi logging calls - intermediate updates to keep the PR sizes down

### CODE REVIEW

File: src/main/kotlin/app/batch/DataKeyProcessor.kt [hunk 1]
```diff
@@ -24,7 +25,7 @@ class DataKeyProcessor(val keyService: KeyService) : ItemProcessor<EncryptedStre
         }
         catch (e: DataKeyDecryptionException) {
             val message = "Failed to decrypt '${item.s3key}': '${e.message}'."
-            logger.error(message)
+            logError(logger, message)
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/DataKeyProcessor.kt]: why the move from `logger.error` to `logError`?

**Comment by dan-hill2802** [comment]
Looks fine if this is the new preferred way of logging. I'm personally not so keen on changing the interface for logging

File: src/main/kotlin/app/batch/DataKeyProcessor.kt (hunk 1 above)
Comment (benj-fletch) [reply src/main/kotlin/app/batch/DataKeyProcessor.kt]: `logError` is the custom method that we are using to format messages as JSON I think

File: src/main/kotlin/app/batch/HBaseWriter.kt [hunk 2]
```diff
@@ -142,12 +145,12 @@ class HBaseWriter : ItemWriter<DecompressedStream> {
                                     if (runMode != RUN_MODE_MANIFEST) {
                                         if (batchSizeBytes + message.length >= maxBatchVolume && batch.size > 0) {
                                             try {
-                                                logger.info("Attempting to write batch of ${batch.size} records, size $batchSizeBytes bytes to hbase with topic 'db.$database.$collection' from '$fileName'.")
+                                                logInfo(logger, "Attempting to write batch of ${batch.size} records, size $batchSizeBytes bytes to hbase with topic 'db.$database.$collection' from '$fileName'.")
                                                 putBatch(batch)
-                                                logger.info("Written batch of ${batch.size} records, size $batchSizeBytes bytes to hbase with topic 'db.$database.$collection' from '$fileName'.")
+                                                logInfo(logger, "Written batch of ${batch.size} records, size $batchSizeBytes bytes to hbase with topic 'db.$database.$collection' from '$fileName'.")
```
Comment (benj-fletch) [inline src/main/kotlin/app/batch/HBaseWriter.kt]: I wonder if having both the pre and post logging lines is a little noisy? I can see the use in summarizing that batches are being / have been written but can't see the benefit of having both over just one?

Comment (benj-fletch) [reply src/main/kotlin/app/batch/HBaseWriter.kt]: Also have just noticed that below we only use a post-write line (line 202) - we should bring these in line

**Review by DanielChicot — APPROVED** [review]

File: src/main/kotlin/app/utils/logging/LoggerUtils.kt [hunk 3]
```diff
@@ -189,6 +155,52 @@ fun getDurationInMilliseconds(epochTime: Long): String {
     return elapsedMilliseconds.toString()
 }
 
+class JsonLoggerWrapper(private val wrappedLogger: Logger) {
```
Comment (benj-fletch) [inline src/main/kotlin/app/utils/logging/LoggerUtils.kt:158]: This class matches a common pattern - `delegate`. Can we rename `wrappedLogger` to `delegate`?

Comment (mark-r-m) [reply src/main/kotlin/app/utils/logging/LoggerUtils.kt:158]: Sure, I'll put that in a following PR.

**Pull request merged**